src/idpi/products/pot_vortic.py

```diff
@@ -0,0 +1,68 @@
+# Standard library
+import sys
+import logging
+
+# Third-party
+import numpy as np
+
+# First-party
+from idpi.operators.destagger import destagger
+from idpi.operators.pot_vortic import fpotvortic
+from idpi.operators.rho import f_rho_tot
+from idpi.operators.theta import ftheta
+from idpi.operators.total_diff import TotalDiff
+from idpi.operators.vertical_interpolation import interpolate_k2p
+from idpi.operators.vertical_interpolation import interpolate_k2theta
+from idpi.operators.vertical_reduction import integrate_k
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def _compute_pot_vortic(U, V, W, T, P, QV, QC, QI, HHL, theta):
+    logger.info("Computing total density")
+    rho_tot = f_rho_tot(T, P, QV, QC, QI)
+
+    logger.info("Computing terrain following grid deformation factors")
+    dlon = HHL.attrs["GRIB_iDirectionIncrementInDegrees"]
+    dlat = HHL.attrs["GRIB_jDirectionIncrementInDegrees"]
+    deg2rad = np.pi / 180
+    total_diff = TotalDiff(dlon * deg2rad, dlat * deg2rad, HHL)
```

Comment:
Note that in our production the post-processing is running in parallel to the model integration, so the potential vorticity operator will be executed each time when a new model output becomes available, and not only at the point when data is available up to the maximum lead time. It should probably be avoided that total_diff will be recomputed every time when a new time step of the model output becomes available for processing. fieldextra actually uses a caching mechanism to avoid re-computation of static information.

Reply:
I agree with that. There is no need to recompute constant data every lead time. Also memory footprint should not be a concern for constant fields.

Reply:
I think that while this is very valuable insight, implementing a caching mechanism to reuse the terrain following terms is out of scope for this PR. Let me open a ticket for this.

Reply:
https://meteoswiss.atlassian.net/browse/APNRZ-111

Reply:
Yes, agree with you, Christian. And we have not yet implemented any time looping nor time operators so far. So temporal aspects should definitely be tacked in separate PRs.